hack/sync-from-kubernetes.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This script is to be used in the interim to pull any new commits relevant to federation in k8s.io/kubernetes.
+# It takes direct inspiration from https://github.com/kubernetes/apimachinery/blob/master/hack/sync-from-kubernetes.sh
+
+set -e
```

Comment:
Instead could use the following
```
set -o errexit
set -o nounset
set -o pipefail
```
This is pretty much the standard followed in kubernetes for all shell scripts.